### PR TITLE
Personal Lists(data-api lists): Zoidberg, Bender and Farnsworth

### PR DESCRIPTION
**Pull request by TawKhoury**
affects: @haaretz/finance, @haaretz/haaretz.co.il, @haaretz/htz-components, @haaretz/app-utils,
@haaretz/graphql-server

**Related issue(s):** #0

## Description

**Notes:**

## Checklist

  * [ ] Approved by designer

Usability:
  * [ ] Developer
  * [ ] Project manager

Tests
  
  * [ ] UI snapshot testing 
  * [ ] Unit testing
  * [ ] Integration testing
  * [ ] E2E testing

### CODE REVIEW

File: packages/components/htz-components/src/components/List/List.js [hunk 1]
```diff
@@ -15,13 +18,36 @@ type ListProps = ListDataType & {
...
+                viewProps={viewProps}
+                ListInner={ListInner}
+                setNoPersonalListData={setNoPersonalListData}
+                noPersonalListData={noPersonalListData}
+              />
+            ) : null}
+          </div>
+        )}
+      </InView>
+    );
```
Comment (TxHawks) [inline packages/components/htz-components/src/components/List/List.js]: Can you extract this to its own component, and dynamically import it?

File: packages/components/htz-components/src/components/List/ListDataGetter.js [hunk 2]
```diff
@@ -109,13 +125,26 @@ function ListRenderer({
     >
       {({ data, loading: listLoading, error: listError, }) => {
         if (listLoading) return <Placeholder />;
-        if (listError) return null;
+        if (listError) {
+          // if no personal data return regular list (load form client)
+          isPersonal && setNoPersonalListData && setNoPersonalListData(true);
+          console.error('!!!! list error: ', listError);
```
Comment (TxHawks) [inline packages/components/htz-components/src/components/List/ListDataGetter.js]: Is this left here by mistake?

File: packages/components/htz-components/src/components/List/List.js [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
 import * as React from 'react';
+import config from 'config';
+import { InView, } from 'react-intersection-observer';
 
 import type { ListDataType, } from '../../flowTypes/ListDataType';
 import type { TeaserDataType, } from '../../flowTypes/TeaserDataType';
 import { getTeasersInPage, updateTeasersInPage, } from './ListDuplication';
 import ReadingHistoryProvider from '../ReadingHistory/ReadingHistoryProvider';
 import useGetComponent from '../../hooks/GetComponentContext/useGetComponent';
+import PersonalList from './PersonalListWrapper';
```
Comment (TxHawks) [inline packages/components/htz-components/src/components/List/List.js]: This should be dynamically imported

Also, why is the import called `PersonalList` and the file `PersonalListWrapper`?

File: packages/libs/graphql-server/src/dataSources.js [hunk 4]
```diff
@@ -606,10 +606,51 @@ class PurchasePageAPI extends RESTDataSource {
...
+    section,
+    // isMobile,
+  }) {
+    // const section = null;
+    const fetchPath = `get-personal-list?listId=${listId
+      || ''}&userId=${userId}&articleId=${articleId}&offset=${offset || 0}&section=${
+      section ? `${section}` : '/'
+    }`;
+
+    console.warn('!!!! fetchPath vars getPersonalList:', fetchPath);
```
Comment (TxHawks) [inline packages/libs/graphql-server/src/dataSources.js]: Left here by mistake?

File: packages/libs/graphql-server/src/dataSources.js [hunk 5]
```diff
@@ -606,10 +606,51 @@ class PurchasePageAPI extends RESTDataSource {
...
+    userId,
+    articleId,
+    // offset,
+    // section
+  }) {
+    // ?userId=1234&articleId=1.8068021&teaserIds=1.234,1.567
+    const fetchPath = `update-reading-history?userId=${userId}&articleId=${articleId}&teaserIds=${teaserIds
+      || null}`;
+
+    console.warn('!!!! updatePersonalList fetchPath:', fetchPath);
```
Comment (TxHawks) [inline packages/libs/graphql-server/src/dataSources.js]: Left in the code by mistake?

File: packages/components/htz-components/src/components/List/ListDataGetter.js (hunk 2 above)
Comment (TxHawks) [inline packages/components/htz-components/src/components/List/ListDataGetter.js]: Left in the code by mistake?

Comment (TawKhoury) [reply packages/components/htz-components/src/components/List/ListDataGetter.js]: console.error has been removed

File: packages/libs/graphql-server/src/dataSources.js (hunk 4 above)
Comment (TawKhoury) [reply packages/libs/graphql-server/src/dataSources.js]: removed

File: packages/libs/graphql-server/src/dataSources.js (hunk 5 above)
Comment (TawKhoury) [reply packages/libs/graphql-server/src/dataSources.js]: removed

File: packages/components/htz-components/src/components/List/List.js (hunk 3 above)
Comment (TawKhoury) [reply packages/components/htz-components/src/components/List/List.js]: Done, PersonalListWrapper is the imort name now

File: packages/components/htz-components/src/components/List/ListDataGetter.js (hunk 2 above)
Comment (TawKhoury) [reply packages/components/htz-components/src/components/List/ListDataGetter.js]: removed

File: packages/components/htz-components/src/components/List/List.js (hunk 1 above)
Comment (TawKhoury) [reply packages/components/htz-components/src/components/List/List.js]: Done

**Pull request merged**